### ISSUE TITLE
Box Scrolling Regressions

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#7037]
[<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5396)*
*Created on Sun May 31 12:44:04 CEST 2026 by [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese)*

---

### Describe the Bug

Some regressions around #7007 (boxes now always scroll)

### To Reproduce

1. Install the UI Inspector add-on
2. Open the UI Inspector and select any child entry
3. See screenshot: All boxes now consist only of scrollbars with the actual content hidden behind them

---

1. Ensure you have some add-ons with long descriptions installed
2. Open Widelands in windowed mode
3. Open the add-ons manager
4. Maximize the main window, then restore it
5. See screenshot: The horizontal box now becomes very wide

### Expected Behavior

None of these cases should have horizontal scrollbars

### Screenshots

![grafik](/attachments/12873ad0-82a5-4725-9a4c-644d4e3002a8)

![grafik](/attachments/4ef5124d-ed5b-410d-888d-3da503538343)

### Operating System

Debian Testing

### Widelands Version

current master @ e0033f7841c3bf16957b0a1460f971a9b092018d

### Enabled Add-Ons

UI Inspector

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5396#issuecomment-16234673)*
*On Sun May 31 15:55:38 CEST 2026, [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) wrote:*

---

I suspect #7027 may also be related. Maybe a fix might be to set (or detect) that certain boxes should never be allowed to scroll?

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5396#issuecomment-16404338)*
*On Tue Jun 02 21:49:43 CEST 2026, [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) wrote:*

---

I think this may be specifically related to another old "friend", the `MultilineTextarea`, which really does not like to reduce its width, so the Box consistently allows it to make itself arbitrarily wide instead of mandating it to use the space the box layout would like to grant it.

https://codeberg.org/wl/widelands/src/commit/3ad6bc7a511a89201d879b0efc3a874f7c4ffa0b/src/ui/basic/multilinetextarea.cc#L128-L133

**Comment by bunnybot** [comment]
Assigned to [<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese) [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) *( `@cb/Nordfriese` )*
